internal/nodes/avi_model_evh_nodes.go

```diff
@@ -666,84 +760,77 @@ func evhNodeHostName(routeIgrObj RouteIngressModel, tlssetting TlsSettings, ingN
 			certsBuilt = true
 		}
 
-		for _, path := range paths {
-			evhNode := vsNode[0].GetEvhNodeForName(lib.GetEvhTlsNodeName(ingName, namespace, evhSecretName, host, path.Path))
-			if evhNode == nil {
-				evhNode = &AviEvhVsNode{
-					Name:         lib.GetEvhTlsNodeName(ingName, namespace, evhSecretName, host, path.Path),
-					VHParentName: vsNode[0].Name,
-					Tenant:       lib.GetTenant(),
-					EVHParent:    false,
-					EvhHostName:  host,
-					EvhPath:      path.Path,
-					ServiceMetadata: avicache.ServiceMetadataObj{
-						NamespaceIngressName: ingressHostMap.GetIngressesForHostName(host),
-						Namespace:            namespace,
-						HostNames:            hosts,
-					},
-				}
-
-				if path.PathType == networkingv1beta1.PathTypeExact {
-					evhNode.EvhMatchCriteria = "EQUALS"
-				} else {
-					// PathTypePrefix and PathTypeImplementationSpecific
-					// default behaviour for AKO set be Prefix match on the path
-					evhNode.EvhMatchCriteria = "BEGINS_WITH"
-				}
-				if lib.GetSEGName() != lib.DEFAULT_SE_GROUP {
-					evhNode.ServiceEngineGroup = lib.GetSEGName()
-				}
-			} else {
-				// The evh node exists, just update the svc metadata
-				evhNode.ServiceMetadata.NamespaceIngressName = ingressHostMap.GetIngressesForHostName(host)
-				evhNode.ServiceMetadata.Namespace = namespace
-				evhNode.ServiceMetadata.HostNames = hosts
-				if evhNode.SSLKeyCertAviRef != "" {
-					certsBuilt = true
-				}
+		evhNode := vsNode[0].GetEvhNodeForName(lib.GetEvhNodeName(ingName, namespace, evhSecretName, host))
+		if evhNode == nil {
+			evhNode = &AviEvhVsNode{
+				Name:         lib.GetEvhNodeName(ingName, namespace, evhSecretName, host),
+				VHParentName: vsNode[0].Name,
+				Tenant:       lib.GetTenant(),
+				EVHParent:    false,
+				EvhHostName:  host,
+				ServiceMetadata: avicache.ServiceMetadataObj{
+					NamespaceIngressName: ingressHostMap.GetIngressesForHostName(host),
+					Namespace:            namespace,
+					HostNames:            hosts,
+				},
 			}
 			if lib.GetSEGName() != lib.DEFAULT_SE_GROUP {
 				evhNode.ServiceEngineGroup = lib.GetSEGName()
 			}
-			evhNode.VrfContext = lib.GetVrf()
-			if !certsBuilt {
-				certsBuilt = aviModel.(*AviObjectGraph).BuildTlsCertNodeForEvh(routeIgrObj.GetSvcLister(), vsNode[0], namespace, tlssetting, key, host)
-			}
-			if certsBuilt {
-				isIngr := routeIgrObj.GetType() == utils.Ingress
-				aviModel.(*AviObjectGraph).BuildPolicyPGPoolsForEVH(vsNode, evhNode, namespace, ingName, key, isIngr, host, path)
-				foundEvhModel := FindAndReplaceEvhInModel(evhNode, vsNode, key)
-				if !foundEvhModel {
-					vsNode[0].EvhNodes = append(vsNode[0].EvhNodes, evhNode)
-				}
+		} else {
+			// The evh node exists, just update the svc metadata
+			evhNode.ServiceMetadata.NamespaceIngressName = ingressHostMap.GetIngressesForHostName(host)
+			evhNode.ServiceMetadata.Namespace = namespace
+			evhNode.ServiceMetadata.HostNames = hosts
+			if evhNode.SSLKeyCertAviRef != "" {
```

Comment:
is this condition true for any EVH node or should be only possible for the parent? In which case, do we need to do a check if it's a parent node?

Reply:
`SSLKeyCertAviRef` is for CRD. In the main code these will take the precedence  over `SSLKeyCertRefs` and will be associated to parent EVH VS.